src/wolfsftp.c

```diff
@@ -1900,13 +1881,10 @@ int wolfSSH_SFTP_RecvOpen(WOLFSSH* ssh, int reqId, byte* data, word32 maxSz)
         return WS_BUFFER_E;
     }
 
-    /* plus one to make sure is null terminated */
-    dir = (char*)WMALLOC(sz + 1, ssh->ctx->heap, DYNTYPE_BUFFER);
-    if (dir == NULL) {
-        return WS_MEMORY_E;
+    if (GetAndCleanPath(ssh->sftpDefaultPath,
```

Comment:
Would it be better to return the error from GetAndCleanPath rather than converting it to WS_BUFFER_E?

Reply:
It would. Changing.